lib/overcommit/git_repo.rb

```diff
@@ -248,5 +248,16 @@ def submodules(options = {})
       raise Overcommit::Exceptions::GitSubmoduleError,
             "Unable to read submodule information from #{ref}:.gitmodules file: #{ex.message}"
     end
+
+    # Returns the names of all branches containing the given commit.
+    #
+    # @param commit_ref [String] git tree ref that resolves to a commit
+    # @return [Array<String>] list of branches containing the given commit
+    def branches_containing_commit(commit_ref)
+      `git branch --contains #{commit_ref}`.
+        split("\n").
+        reject { |s| s[/detached from/] }.  # ignore detached HEAD
+        map { |s| s[2..-1] }                # trim leading 2 characters
```

Comment:
Sadly, the output of `git branch` can change based on the user's `column.branch` setting, so we need to specify the `--column` flag explicitly to override whatever that is set to.

After playing around with a few of the options the best output I could find was:

```
git branch --contains <hash> --column=dense`
```

...which outputs:

```
  master * current-branch some/other/branch another-branch
```

Since the asterisk marks the current branch and branches can't have spaces, we could `split(/\s+/)` and filter out the asterisk to get all branches.